advisories/a93398.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Technical Advisory 93398
+advisory: A-93398
+summary: Altering a non-empty table to add a column with a <code>DEFAULT</code> expression in which the type of the expression did not match the type of the new column could cause the column to become corrupted.
+toc: true
+affected_versions: v22.2.0-alpha.1 to v22.2.3
+advisory_date: 2023-03-01
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+In all versions of CockroachDB from [v22.2.0-alpha.1](../releases/v22.2.html#v22-2-0-alpha-1) to [v22.2.3](../releases/v22.2.html#v22-2-3), altering a non-empty table to [add a column](../v22.2/alter-table.html#add-column) with a [`DEFAULT` expression](../v22.2/default-value.html) in which the type of the expression does not match the type of the new column could cause the column to become corrupted. For example, `ALTER TABLE t ADD COLUMN c DECIMAL DEFAULT pi();`. This happens because the `DEFAULT` expression was not cast to the correct type when the column was backfilled for the existing rows in the table. As a result, a later attempt to read the data in that column could cause the decoding logic to panic and crash the node.
```

Comment:
nit: the tenses are inconsistent -- I'd say either "happens because .... is not cast ... when the column is backfilled" or "happened because ... was not cast ... when the column was backfilled""

Reply:
Thanks for catching, I switched to present here since we're talking about what happens in the versions affected!